### PR TITLE
[SCB-2380] optimize swaager load logic

### PR DESCRIPTION
**Pull request by david6969xin**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/AbstractSwaggerGenerator.java [hunk 1]
```diff
@@ -311,6 +314,21 @@ protected void scanMethods() {
...
+        originsDefinitions.forEach((key, value) -> {
+          if (definitions.containsKey(key) && !value.equals(definitions.get(key))) {
+            throw new IllegalArgumentException(
+                "the parameter typeName '" + key + "' duplicate in '" + this.cls.getName()
+                    + "', need to rename it");
+          }
+        });
+        definitions.putAll(originsDefinitions);
+      }
+      swagger.setDefinitions(null);
```
Comment (liubao68) [inline swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/AbstractSwaggerGenerator.java:328]: It's a bit complicated. 

Maybe you can have a very simple solution to extend Swagger `public void addDefinition(String key, Model model)` method

**Comment by david6969xin** [comment]
retrigger cl

**Comment by david6969xin** [comment]
retrigger cl

**Comment by david6969xin** [comment]
retrigger cl

**Comment by david6969xin** [comment]
retrigger cl

File: swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/AbstractSwaggerGenerator.java [hunk 2]
```diff
@@ -81,9 +84,25 @@
...
+  class SwaggerExt extends Swagger {
+    @Override
+    public void addDefinition(String key, Model model) {
+      if (this.definitions == null) {
+        this.definitions = new LinkedHashMap<>();
+      }
+      if (this.definitions.containsKey(key) && !this.definitions.get(key).equals(model)) {
+        throw new IllegalArgumentException(
+            "the parameter typeName " + key + " duplicate in " + cls.getName()
+                + ", need to rename it");
```
Comment (liubao68) [inline swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/AbstractSwaggerGenerator.java:97]: throw new IllegalArgumentException("Model name can not be same for different class. Name=“ + key + ", Class=" + cls.getName);

File: swagger/swagger-generator/generator-core/src/test/java/org/apache/servicecomb/swagger/generator/core/TestSwaggerUtils.java [hunk 3]
```diff
@@ -247,7 +247,7 @@ public void should_ignore_httpServletRequest() {
   @Test
   public void testRepeatOperation() {
     UnitTestSwaggerUtils.testException(
-        "OperationId must be unique. method=org.apache.servicecomb.swagger.generator.core.schema.RepeatOperation:add.",
+        "generate swagger operation failed, method=org.apache.servicecomb.swagger.generator.core.schema.RepeatOperation:add.",
```
Comment (liubao68) [inline swagger/swagger-generator/generator-core/src/test/java/org/apache/servicecomb/swagger/generator/core/TestSwaggerUtils.java:250]: Why this test case fail? Seam not related to this PR.

**Comment by liubao68** [comment]
This improvement may cause other more problems, do not fix it.

**Pull request closed without merging**